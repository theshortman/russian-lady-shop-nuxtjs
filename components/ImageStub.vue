<template>
  <svg
    class="img-fluid img-thumbnail"
    width="600"
    height="912"
    xmlns="http://www.w3.org/2000/svg"
    role="img"
    aria-label="No photo"
    preserveAspectRatio="xMidYMid slice"
    focusable="false"
  >
    <rect width="100%" height="100%" fill="#ffffff"></rect>
    <text x="50%" y="50%" fill="#dee2e6" dy=".3em" text-anchor="middle">
      {{ text }}
    </text>
  </svg>
</template>

<script>
export default {
  props: {
    text: {
      type: String,
      default: null,
    },
  },
}
</script>
