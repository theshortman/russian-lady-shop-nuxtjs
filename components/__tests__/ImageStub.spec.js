import { shallowMount } from '@vue/test-utils'
import ImageStub from '../ImageStub.vue'

describe('ImageStub.vue', () => {
    test('renders correctly', () => {
        const wrapper = shallowMount(ImageStub)
        
        expect(wrapper.html()).toMatchSnapshot()
    })

    test('renders correct text', () => {
        const text = 'test text'

        const wrapper = shallowMount(ImageStub, {
            propsData: {
                text,
            }
        })

        expect(wrapper.text()).toBe(text)
    })
})