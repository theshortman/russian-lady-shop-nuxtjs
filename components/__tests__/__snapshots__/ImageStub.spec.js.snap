// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ImageStub.vue renders correctly 1`] = `
"<svg width=\\"600\\" height=\\"912\\" xmlns=\\"http://www.w3.org/2000/svg\\" role=\\"img\\" aria-label=\\"No photo\\" preserveAspectRatio=\\"xMidYMid slice\\" focusable=\\"false\\" class=\\"img-fluid img-thumbnail\\">
  <rect width=\\"100%\\" height=\\"100%\\" fill=\\"#ffffff\\"></rect> <text x=\\"50%\\" y=\\"50%\\" fill=\\"#dee2e6\\" dy=\\".3em\\" text-anchor=\\"middle\\">

  </text>
</svg>"
`;
